src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -221,9 +224,23 @@ export function DiscoverMainRoute({ customizationCallbacks, isDev }: MainRoutePr
 
   // primary fetch: on initial search + triggered when id changes
   useEffect(() => {
+    setLoading(true);
+    setHasESData(false);
+    setHasUserDataView(false);
+    setShowNoDataPage(false);
+    setError(undefined);
```

Comment:
QQ: do we need to set `setHasESData`, `setHasUserDataView`, `setShowNoDataPage`? this would cover the case that a user already loaded a saved search (which means those value have already been set), but then the data + data views were deleted

Reply:
@kertal It's to reset the state fully - more predictable this way. Also this PR allows to bypass the empty screen when opening a saved search. But the saved can be missing. So when we guide the user to go back, checks need to happen again.

Reply:
makes sense, thx!